src/v/cluster/shard_table.h

```diff
@@ -26,9 +27,9 @@ namespace cluster {
 class shard_table final {
     struct shard_revision {
         ss::shard_id shard;
-        model::revision_id revision;
         /// Only used for additional sanity checks
         bool non_replicable;
+        model::revision_id revision;
```

Comment:
Save 8 bytes by reordering, this is a hot structure so it is worth it I think.

Reply:
Yes exactly. Groups like-sized fields together so that padding is minimized. There's a tool `pahole` to identify places to do it automatically, might be interesting to run it on our stuff (not sure if it handles C++). 

Sometimes there is a tension between usability when we have plain aggregates where the stuff has a "logical" order which conflicts with the ideal order from a size PoV but that didn't seem to be the case here.

Reply:
cool, thanks for the reference to pahole